src/Compression/Compression.cs

```diff
@@ -29,12 +32,41 @@ public partial class CompressionStream : Stream
 	{
 		private const int DefaultBufferSize = 8192;
 
-		private Stream _stream;
+		private Stream? _stream;
+		Stream Stream {
+			get {
+				if (_stream is null)
+					throw new ObjectDisposedException (null, "Can not access a closed Stream.");
+				return _stream;
+			}
+		}
 		private CompressionMode _mode;
 		private bool _leaveOpen;
-		private Inflater _inflater;
-		private Deflater _deflater;
-		private byte[] _buffer;
+		private Inflater? _inflater;
+		Inflater Inflater {
+			get {
+				if (_inflater is null)
+					_inflater = new Inflater (_algorithm);
+				return _inflater;
+			}
+		}
+		private Deflater? _deflater;
+		Deflater Deflater {
+			get {
+				if (_deflater is null)
+					_deflater = new Deflater (_algorithm);
+				return _deflater;
+			}
+		}
+		private byte[]? _buffer;
+		Byte[] Buffer {
+			get {
+				if (_buffer is null)
+					_buffer = ArrayPool<byte>.Shared.Rent (DefaultBufferSize);
```

Comment:
```suggestion
					throw new InvalidOperationException ("Buffer has not been initialized");
```

and keep `EnsureBufferInitialized`.

Automatically initializing can sometimes run into problems where the field is cleared, and then inadvertently	re-created, which causes for some hard-to-find bugs. It's easier if an exception is thrown in that scenario.